### PR TITLE
Memory probing via grub

### PR DESCRIPTION
**Pull request by bstrzelecki**
Aditional build dependencies:
- Grub
- Xorriso
- mtools

### CODE REVIEW

**Review by spohie001 — APPROVED** [review]
(o゜▽゜)o☆

File: kernel/kernel/memory.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <kernel/memory.h>
+
+struct{
+	void* base;
+	size_t length;
+} avaiable_memory;
```
Comment (BognaLew) [inline kernel/kernel/memory.c]:
```suggestion
} available_memory;
```

File: kernel/kernel/memory.c [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+void init_memory_manager(void* base, size_t length){
+	avaiable_memory.base = base;
```
Comment (BognaLew) [inline kernel/kernel/memory.c]:
```suggestion
	available_memory.base = base;
```

File: kernel/kernel/memory.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+	avaiable_memory.length = length;
```
Comment (BognaLew) [inline kernel/kernel/memory.c]:
```suggestion
	available_memory.length = length;
```

**Review by BognaLew — APPROVED** [review]

**Pull request merged**